ExampleMod/Common/UI/ExampleDisplaySets/ExampleReversedBarsDisplay.cs

```diff
@@ -0,0 +1,203 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria;
+using Terraria.GameContent.UI.ResourceSets;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common.UI.ExampleDisplaySets
+{
+	// This file showcases an example resource display set
+	// In this example, it draws a clone of the Bars display set, but with the mana and stars "swapped" to the other location
+	public class ExampleReversedBarsDisplay : ModResourceDisplaySet
+	{
+		public static readonly ResourceSetSlotId<ExampleReversedBarsDisplay> ManaPanels = 0;
+		public static readonly ResourceSetSlotId<ExampleReversedBarsDisplay> ManaBars = 1;
+		public static readonly ResourceSetSlotId<ExampleReversedBarsDisplay> LifePanels = 2;
+		public static readonly ResourceSetSlotId<ExampleReversedBarsDisplay> LifeBars = 3;
+
+		// Variable names were copied from HorizontalBarsPlayerResourcesDisplaySet
+		private int _maxSegmentCount;
+		private int _hpSegmentsCount;
+		private int _mpSegmentsCount;
+		private int _hpFruitCount;
+		private float _hpPercent;
+		private float _mpPercent;
+		private bool _hpHovered;
+		private bool _mpHovered;
+		private Asset<Texture2D> _hpFill;
+		private Asset<Texture2D> _hpFillHoney;
+		private Asset<Texture2D> _mpFill;
+		private Asset<Texture2D> _panelLeft;
+		private Asset<Texture2D> _panelMiddleHP;
+		private Asset<Texture2D> _panelRightHP;
+		private Asset<Texture2D> _panelMiddleMP;
+		private Asset<Texture2D> _panelRightMP;
+
+		private PlayerStatsSnapshot preparedSnapshot;
+
+		public override void Load() {
+			string vanillaFolder = "Images/UI/PlayerResourceSets/HorizontalBars/";
+			string modFolder = "ExampleMod/Common/UI/ExampleDisplaySets/";
+			_hpFill = Main.Assets.Request<Texture2D>(vanillaFolder + "HP_Fill", AssetRequestMode.ImmediateLoad);
+			_hpFillHoney = Main.Assets.Request<Texture2D>(vanillaFolder + "HP_Fill_Honey", AssetRequestMode.ImmediateLoad);
+			_mpFill = Main.Assets.Request<Texture2D>(vanillaFolder + "MP_Fill", AssetRequestMode.ImmediateLoad);
+
+			_panelLeft = Main.Assets.Request<Texture2D>(vanillaFolder + "Panel_Left", AssetRequestMode.ImmediateLoad);
+			_panelMiddleHP = Main.Assets.Request<Texture2D>(vanillaFolder + "HP_Panel_Middle", AssetRequestMode.ImmediateLoad);
+			_panelRightHP = ModContent.Request<Texture2D>(modFolder + "HP_Panel_Right", AssetRequestMode.ImmediateLoad);
+			_panelMiddleMP = Main.Assets.Request<Texture2D>(vanillaFolder + "MP_Panel_Middle", AssetRequestMode.ImmediateLoad);
+			_panelRightMP = ModContent.Request<Texture2D>(modFolder + "MP_Panel_Right", AssetRequestMode.ImmediateLoad);
+		}
+
+		// DrawLife runs before DrawMana, so keep that in mind
+		// In this case, the bars are swapped, so the order of drawing (top bar, then bottom bar) should be preserved
+		// This is why Mana resources are being drawn in DrawLife and Life resources are being drawn in DrawMana
```

Comment:
> why must the order be preserved (top bar then bottom bar)?

otherwise, the bottom bar would clip into the top bar, which wouldn't be consistent with how the original display that this was "cloned" from works

Reply:
if it would make sense for the life bar to still draw below the mana bar, then i'll change this to reflect that

Reply:
fair enough.  i'll swap the two method bodies